src/System.Net.Http/src/uap/System/Net/HttpHandlerToFilter.cs

```diff
@@ -26,65 +26,207 @@ namespace System.Net.Http
-            {
-                throw;
-            }
-            catch (Exception)
+            while (true)
             {
-                object info;
-                if (rtRequest.Properties.TryGetValue(SavedExceptionDispatchInfoLookupKey, out info))
+                cancel.ThrowIfCancellationRequested();
+
+                if (response != null)
                 {
-                    ((ExceptionDispatchInfo)info).Throw();
+                    response.Dispose();
+                    response = null;
                 }
 
-                throw;
-            }
+                RTHttpRequestMessage rtRequest = await ConvertRequestAsync(
+                    request,
+                    requestHttpMethod,
+                    skipRequestContentIfPresent).ConfigureAwait(false);
+
+                RTHttpResponseMessage rtResponse;
+                try
+                {
+                    if (redirects > 0)
+                    {
+                        rtResponse = await _filterWithNoCredentials.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        rtResponse = await _filter.SendRequestAsync(rtRequest).AsTask(cancel).ConfigureAwait(false);
+                    }
+                }
+                catch (TaskCanceledException)
+                {
+                    throw;
+                }
+                catch (Exception)
+                {
+                    object info;
+                    if (rtRequest.Properties.TryGetValue(SavedExceptionDispatchInfoLookupKey, out info))
+                    {
+                        ((ExceptionDispatchInfo)info).Throw();
+                    }
+
+                    throw;
+                }
+
+                response = ConvertResponse(rtResponse);
+
+                ProcessResponseCookies(response, request.RequestUri);
+
+                if (!_handler.AllowAutoRedirect)
+                {
+                    break;
+                }
+
+                if (response.StatusCode != HttpStatusCode.MultipleChoices &&
+                    response.StatusCode != HttpStatusCode.MovedPermanently &&
+                    response.StatusCode != HttpStatusCode.Redirect &&
+                    response.StatusCode != HttpStatusCode.RedirectMethod &&
+                    response.StatusCode != HttpStatusCode.RedirectKeepVerb)
+                {
+                    break;
+                }
+
+                redirects++;
+                if (redirects > _handler.MaxAutomaticRedirections)
+                {
+                    break;
+                }
+
+                Uri redirectUri = response.Headers.Location;
+                if (redirectUri == null)
+                {
+                    break;
+                }
+
+                if (!redirectUri.IsAbsoluteUri)
+                {
+                    redirectUri = new Uri(string.Format("{0}://{1}:{2}{3}",
```

Comment:
Update: I checked again and I can see that ctor overload.  I changed that code and tests still pass.

```c#
if (!redirectUri.IsAbsoluteUri)
{
    redirectUri = new Uri(request.RequestUri, redirectUri.OriginalString);
}
```